game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -613,20 +615,20 @@ private void removeTerritoriesThatArentWorthAttacking(
 
         // Determine if any of the attacking from territories has enemy neighbors that aren't being
         // attacked
-        boolean attackersHaveEnemyNeighbors = false;
-        Territory attackFromTerritoryWithEnemyNeighbors = null;
-        for (final Territory attackFromTerritory : attackFromTerritories) {
-          final Set<Territory> enemyNeighbors =
-              data.getMap()
-                  .getNeighbors(
-                      attackFromTerritory, ProMatches.territoryIsEnemyNotNeutralLand(player));
-          if (!prioritizedTerritoryList.containsAll(enemyNeighbors)) {
-            attackersHaveEnemyNeighbors = true;
-            attackFromTerritoryWithEnemyNeighbors = attackFromTerritory;
-            break;
-          }
-        }
-        if (attackersHaveEnemyNeighbors) {
+        // Note: Use territoryIsEnemyNotNeutralLand(), not territoryIsEnemyNotPassiveNeutralLand()
+        // so that the neutrality check is consistent with logic for isNeutral.
+        Predicate<Territory> enemyTerritory = ProMatches.territoryIsEnemyNotNeutralLand(player);
+        Territory attackFromTerritoryWithEnemyNeighbors =
```

Comment:
This is the same logic as before, just using stream() API. The functional change here is the predicate used above, see Note: comment.